include/ur_ddi.h

```diff
@@ -1485,6 +1701,46 @@ typedef ur_result_t(UR_APICALL *ur_pfnGetUSMProcAddrTable_t)(
     ur_api_version_t,
     ur_usm_dditable_t *);
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urUSMPitchedAllocExp
+typedef ur_result_t(UR_APICALL *ur_pfnUSMPitchedAllocExp_t)(
+    ur_context_handle_t,
```

Comment:
I don't think that you need the `ur_context_handle_t` argument, you get the native `cuContext` from the device argument. I think this is true of all adapters that will support this feature.
I don't think unused arguments should be included in the interfaces.

Reply:
I may be wrong but I think its included to enable the future work of supporting Level Zero.

Reply:
Just looked at L0 PI and you're likely right (Although I'm not certain in this specific case).
They use context only to prevent sycl UB of devices in different contexts copying memory to each other.
Point is l0 plugin seems to be a lot stricter on this than the last time I looked at it. Since level_zero seems to have fully adopted this context constraint probably there is nothing we can do about it now, even though noone ever came up with a use case for introducing this constraint (which really defines context) in any of these backends.

Reply:
L0 usm allocation functions appear to take and use a context argument. So I guess they will expect this new alloc function to also have the context.